group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -998,4 +1055,93 @@ public ConsumerGroupDescribeResponseData.DescribedGroup asDescribedGroup(
         );
         return describedGroup;
     }
+
+
+    /**
+     * Set the attributes of the consumer group according to a classic group.
+     * Add the records for creating and updating the consumer group.
+     *
+     * @param classicGroup      The converted classic group.
+     * @param records           The list to which the new records are added.
+     */
+    public void fromClassicGroup(
+        ClassicGroup classicGroup,
+        List<Record> records,
+        TopicsImage topicsImage
+    ) {
+        setGroupEpoch(classicGroup.generationId());
+        records.add(RecordHelpers.newGroupEpochRecord(groupId(), classicGroup.generationId()));
+        // SubscriptionMetadata will be computed in the following consumerGroupHeartbeat.
+        records.add(RecordHelpers.newGroupSubscriptionMetadataRecord(groupId(), Collections.emptyMap()));
+
+        setTargetAssignmentEpoch(classicGroup.generationId());
+        records.add(RecordHelpers.newTargetAssignmentEpochRecord(groupId(), classicGroup.generationId()));
+
+        classicGroup.allMembers().forEach(member -> {
+            try {
+                ConsumerPartitionAssignor.Assignment assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(member.assignment()));
+                Map<Uuid, Set<Integer>> partitions = topicPartitionMapFromList(assignment.partitions(), topicsImage);
+                ConsumerPartitionAssignor.Subscription subscription = ConsumerProtocol.deserializeSubscription(ByteBuffer.wrap(member.metadata(classicGroup.protocolName().get())));
+
+                ConsumerGroupMember newMember = new ConsumerGroupMember.Builder(member.memberId())
+                    .setMemberEpoch(classicGroup.generationId())
+                    .setPreviousMemberEpoch(classicGroup.generationId())
+                    .setInstanceId(member.groupInstanceId().orElse(null))
+                    .setRackId(subscription.rackId().orElse(null))
+                    .setRebalanceTimeoutMs(member.rebalanceTimeoutMs())
+                    .setClientId(member.clientId())
+                    .setClientHost(member.clientHost())
+                    .setSubscribedTopicNames(subscription.topics())
+                    .setAssignedPartitions(partitions)
+                    .setSupportedProtocols(member.supportedProtocols())
+                    .build();
+                updateMember(newMember);
+
+                records.add(RecordHelpers.newMemberSubscriptionRecord(groupId(), newMember));
+                records.add(RecordHelpers.newCurrentAssignmentRecord(groupId(), newMember));
+                records.add(RecordHelpers.newTargetAssignmentRecord(groupId(), member.memberId(), partitions));
+            } catch (SchemaException e) {
+                log.warn("Failed to parse Consumer Protocol " + ConsumerProtocol.PROTOCOL_TYPE + ":" +
+                    classicGroup.protocolName().get() + " of group " + groupId + ".", e);
```

Comment:
Shouldn't we throw an exception to fail the request too? It could be a `GroupIdNotFoundException` with an appropriate error message.